### PR TITLE
ExecuteScriptCommand

### PR DESCRIPTION
**Pull request by mateon1**

### CODE REVIEW

File: src/ground_script_1.c [hunk 1]
```diff
@@ -0,0 +1,1863 @@
...
+extern char gUnknown_8116644[];
+extern char gUnknown_8116664[];
+extern char gUnknown_8116684[];
+extern char gUnknown_81166C0[];
+extern char gUnknown_81166D8[];
+
+
+extern DebugLocation gUnknown_81166B4;
+
+/* No decompiler yet implemented for arm32 */
```
Comment (AnonymousRandomPerson) [inline src/ground_script_1.c:212]: Do we need this comment? It looks like just the comment decomp.me adds into new scratches.

File: src/ground_script_1.c [hunk 2]
```diff
@@ -0,0 +1,1863 @@
...
+                }
+                break;
+            }
+            case 0x1f: {
+                s32 a = (s16)sub_8001658(0, 19);
+                const DungeonInfo *ret1 = sub_80A2608(a);
+                s32 thing = sub_8001784(0, 48, a) == 0 ? ret1->unk6 : ret1->unk8;
+                //s32 as = a;
+                //if (!(a >= 0x37 && a < 0x48) && (s16)sub_80A2750(a) == 1) {
+                // fakematch lmao
```
Comment (AnonymousRandomPerson) [inline src/ground_script_1.c:540]: Is this still a fakematch? If not then this commented-out code isn't needed anymore.

File: src/ground_script_1.c [hunk 3]
```diff
@@ -0,0 +1,1863 @@
...
+                if (curCmd.argByte) return 2;
+                break;
+            }
+            case 0x26: {
+                sub_8099A48(curCmd.argShort);
+                if (curCmd.argByte) return 2;
+                break;
+            }
+            case 0x27: case 0x28: {
+                //s8 color[4];
```
Comment (AnonymousRandomPerson) [inline src/ground_script_1.c:635]: Remove commented-out code.

File: src/ground_script_1.c [hunk 4]
```diff
@@ -0,0 +1,1863 @@
...
+void GroundLives_CancelBlank_1(void);
+void GroundObject_CancelBlank(void);
+void GroundEffect_CancelBlank(void);
+void GroundWeather_Select(s16);
+u32 GroundLives_ExecutePlayerScriptActionLives();
+s16 GroundObject_Add(s16 id, GroundObjectData*, s16 group, s8 sector);
+s16 GroundEffect_Add(s16 id, GroundEffectData*, s16 group, s8 sector);
+u32 FlagCalc();
+u32 FlagJudge();
+s32 OtherRandInt(s32); // IT RETURNS s32!! Not s16!
```
Comment (SethBarberee) [inline src/ground_script_1.c:65]: That's the current definition in `other_random.h` so should be good to use that instead.

File: src/ground_script_1.c [hunk 5]
```diff
@@ -0,0 +1,1863 @@
...
+u32 sub_8002318();
+u32 sub_80023E4();
+u32 sub_80026CC();
+void sub_80026E8(s16, bool8);
+char sub_8002984(s32, u8);
+u32 sub_8002C60();
+u32 sub_8002D54();
+u8 sub_8002DF0(Position32*, Position32*, Position32*, Position32*);
+s32 sub_8009FB8();
+u32 StopAllMusic_1();
```
Comment (SethBarberee) [inline src/ground_script_1.c:92]: Can probably import `code_80118A4.h` for all of these.

File: src/ground_script_1.c [hunk 6]
```diff
@@ -0,0 +1,1863 @@
...
+u32 xxx_call_stop_fanfare_se();
+u32 xxx_call_fade_out_fanfare_se();
+void xxx_call_fade_in_new_bgm(u16 id, u16 arg); //sub_8011900
+bool8 sub_8021700(s32);
+bool8 sub_802FCF0(void);
+
+
+s32 FindItemInInventory();
+u32 ShiftItemsDownFrom();
+u32 sub_80920B8();
```
Comment (SethBarberee) [inline src/ground_script_1.c:108]:
```suggestion
void sub_80920B8(u8 *buffer)
```

From `rescue_team.h`

File: src/ground_script_1.c [hunk 7]
```diff
@@ -0,0 +1,1863 @@
...
+            case 0xee: { // RESTORE_SAVED
+                if (scriptData->savedScript.ptr == NULL) {
+                    return 0; // huh?
+                }
+                scriptData->script = scriptData->savedScript;
+                scriptData->savedScript.ptr = NULL;
+                scriptData->savedScript.ptr2 = NULL;
+                break;
+            }
+            case 0xef: // RETURN_FALSE
```
Comment (SethBarberee) [inline src/ground_script_1.c:1851]: Is it worth it to create an enum for all these cases? I recall you have extensive notes on each case plus what's available online.

Comment (mateon1) [reply src/ground_script_1.c:1851]: Well, I don't have a complete description of all cases, so most enum variants would be `SCRIPT_CMD_6C` and such. And besides, I'm pretty sure they used raw integers here (see case ranges).
We can define macros for script commands when defining scripts in data/*.

File: src/ground_script_1.c (hunk 2 above)
Comment (mateon1) [reply src/ground_script_1.c:540]: I'm pretty sure this is still a fakematch, this is a range check like `0x37 <= a && a < 0x48` of some sort (which generates the correct code minus u32->u16 cast). I think an inline might be involved?

File: src/ground_script_1.c (hunk 5 above)
Comment (mateon1) [reply src/ground_script_1.c:96]: Unfortunately not all of these, left a comment that they should probably be moved into the header file though

**Review by SethBarberee — APPROVED** [review]
🫡 He does it again. Looks good for now.

**Review by AnonymousRandomPerson — APPROVED** [review]

**Pull request merged**